mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp

```diff
@@ -3865,27 +3923,13 @@ struct Conv1DGenerator
         oper = Pool;
         isPoolExt = true;
         poolExtOp = feedOp->getName().getIdentifier();
-      } else if (!((isa<arith::MulIOp, arith::MulFOp>(feedOp) ||
-                    (isa<arith::AndIOp>(feedOp) &&
-                     feedOp->getResultTypes()[0].isInteger(1))) &&
-                   llvm::all_of(feedOp->getOperands(), [](Value v) {
-                     if (isa<BlockArgument>(v))
-                       return true;
-                     if (Operation *op = v.getDefiningOp())
-                       return isCastOfBlockArgument(op);
-                     return false;
-                   }))) {
-        return false;
+      } else {
+        oper = Conv;
```

Comment:
Style nit: prefer early exit and continue to simplify the code. (IMO, it also saves the level of indents, which looks better to me.)

```cpp
if (numBlockArguments == 1) {
  // ...
  if (isCastOfBlockArgument(feedOp)) {
    oper = Pool;
    isPoolExt = true;
    poolExtOp = feedOp->getName().getIdentifier();
    return;
  }
  oper = Conv;
  return;
}
oper = Pool;
isPoolExt = false;
```

https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code

Reply:
This does make it cleaner.